planner/core/point_get_plan.go

```diff
@@ -1006,3 +1042,73 @@ func (p *PointGetPlan) findHandleCol() *expression.Column {
 	}
 	return handleCol
 }
+
+func buildSchemaForPartition(dbName model.CIStr, tbl *model.TableInfo, tblName model.CIStr) (*expression.Schema, []*types.FieldName) {
+	columns := make([]*expression.Column, 0, len(tbl.Columns)+1)
+	names := make([]*types.FieldName, 0, len(tbl.Columns)+1)
+	for _, col := range tbl.Columns {
+		names = append(names, &types.FieldName{
+			DBName:      dbName,
+			OrigTblName: tbl.Name,
+			TblName:     tblName,
+			ColName:     col.Name,
+		})
+		columns = append(columns, colInfoToColumn(col, len(columns)))
+	}
+	return expression.NewSchema(columns...), names
+}
+
+func locateHashPartition(piExpr expression.Expression, pairs []nameValuePair) (int64, bool) {
```

Comment:
How much faster does this compares to `partitionedTable.locateHashPartition`?

Reply:
current implement may be faster, but i think `partitionedTable.locateHashPartition` is easier to understand. I will use `partitionedTable.locateHashPartition` to replace it later.

Reply:
I think we better only support `Column` expression for now.
Other types are rarely used.

If we support other types, we need to add test cases for them.